FULL_HELP_DOCS.md

```diff
@@ -1089,17 +1090,18 @@ By default, when starting a testnet container, without any optional arguments, i
 
 Stop a network started with `network start`. For example, if you ran `stellar network start local`, you can use `stellar network stop local` to stop it.
 
-**Usage:** `stellar network stop [OPTIONS] <NETWORK>`
+**Usage:** `stellar network stop [OPTIONS] [NETWORK]`
 
 ###### **Arguments:**
 
-* `<NETWORK>` — Network to stop
+* `<NETWORK>` — Network to stop (used in container name generation)
 
   Possible values: `local`, `testnet`, `futurenet`, `pubnet`
 
 
 ###### **Options:**
 
+* `-c`, `--container-name <CONTAINER_NAME>` — Optional argument to specify the container name
```

Comment:
Having both network and name on the creation command makes sense, having both on other commands is confusing. If I create a container like so:

```
$ stellar network start testnet --name amazetestnet
```

Why do I need to specify anything other than the name at shutdown?

```
$ stellar network stop amazetestnet
```

The stop command shouldn't need to know what type of network is being shutdown, just it's identifier/name.

If we're adding configurable names, I think the other commands should have `<NETWORK>` replaced with `<NAME>`. E.g.:

```
stellar network stop [OPTIONS] [NAME]
```

Reply:
I agree. I think we originally liked the ability to have `network container start <NETWORK>` and `network container stop <NETWORK>` being the same signature. But now that we're adding the ability to change the container name, that probably makes less sense, and adds confusion.

Reply:
I also think that we were allowing for the user not knowing what the container name was and still being able to stop it. But now that we're printing out the full stop and log commands in the `start` command output, I think this is overkill and not necessary.

Reply:
I just pushed up a commit to simplify this so that it can be

```
stellar network start testnet --name tn1
```
and 
```
$ stellar network logs tn1
```
```
$ stellar network stop tn1
```


However, I'm realizing that this is technically a breaking change since we're changing the way that `stop` and `logs` are called. Is this something we want to wait on?

Reply:
Yeah I think we want to hold off all breaking changes to protocol boundaries. Let's wait when CLI upgrades to v22.

Reply:
Sounds good - I'll revert this for now. I've also created an issue so that we can handle this at the next protocol release: https://github.com/stellar/stellar-cli/issues/1463

Reply:
I don't think this is a breaking change because today the names are not configurable, and the network type is the name, the commands already appear to accept the names.

Today, where all commands accept the network, this is possible:

```
$ stellar network start testnet
$ stellar network logs testnet
$ stellar network stop testnet
```

Tomorrow, where other commands accept the name, this is possible:

```
$ stellar network start testnet
$ stellar network logs testnet
$ stellar network stop testnet
$ stellar network start testnet --name tn1
$ stellar network logs tn1
$ stellar network stop tn1
```

When no --name parameter is passed, the command sequence is the same and unbroken.

Reply:
In the tomorrow scenario above:

When no `--name` param is passed to `stellar network container start testnet`, the container that is started is currently called `stellar-testnet`.

In order to stop the container or get the logs, the user would have to do:
```
$ stellar network container logs stellar-testnet
$ stellar network container stop stellar-testnet
```

This would be slightly different than their existing workflow... because they'd need to pass in the container name rather than the network.

Perhaps this is identifying that defaulting the container name to just the network is what we're after, instead of prepending it with `stellar-`. That would likely simplify things.

Reply:
Ah, I hadn't noticed the stellar- prefix. I do think the stellar prefix is helpful, and maybe it should always be present even when a custom name is provided?

Reply:
Or remove the stellar- prefix, either approach sounds reasonable to me.

Reply:
Thinking about this more - if I do make the change to default the container to the network name, and remove the optional network arg for `stop` and `logs`, then we would have these scenarios as you mentioned above @leighmcculloch. I like it, and am going to spike it out to see how much it simplifies things.

today:
```
$ stellar network container start testnet
$ stellar network container logs testnet
$ stellar network container stop testnet
```

tomorrow:
```
$ stellar network container start testnet
$ stellar network container logs testnet
$ stellar network container stop testnet
$ stellar network container start testnet --name tn1
$ stellar network container logs tn1
$ stellar network container stop tn1
```

Reply:
I didn't see your message earlier - looks like we posted very close to the same time! 

But yeah, I just pushed a change removing the `stellar-` prefix. I also thought it was helpful to have it... but I am thinking that if a user wants it, they can always use the `--name` flag!

Reply:
Giving this some more thought and I think we might regret removing the prefix. Namespacing the containers the CLI is using is helpful, for examples for folks who use docker for other things. It keeps the containers that the stellar CLI somewhat namespaced. 

Can we keep the prefix and apply it to the --name option too? So if you name a network container `tn1` it's docker container name is `stellar-tn1`.

Reply:
I was imagining with this change that the name of the container from the user pov would be without the stellar- prefix, and the stellar- prefix would be something the CLI added for the docker container.

e.g.:
```
$ stellar network container start testnet
$ docker ps -l
CONTAINER ID   IMAGE                         NAMES
56eb4d93c444   stellar/quickstart:testing    stellar-testnet
$ stellar network container logs testnet
$ stellar network container stop testnet

$ stellar network container start testnet --name tn1
$ docker ps -l
CONTAINER ID   IMAGE                         NAMES
c3f279116899   stellar/quickstart:testing    stellar-tn1
$ stellar network container logs tn1
$ stellar network container stop tn1
```

Reply:
Ah, I gotcha! I just pushed an update.